src/ConnectionAutomaton.h

```diff
@@ -30,8 +32,10 @@ using StreamId = uint32_t;
 /// Returns true if the responder has been created successfully, false if the
 /// frame cannot start a new stream, in which case the frame (passed by a
 /// mutable referece) must not be modified.
-using StreamAutomatonFactory =
-    std::function<bool(StreamId, std::unique_ptr<folly::IOBuf>)>;
+using StreamAutomatonFactory = std::function<bool(
+    ConnectionAutomaton& connection,
```

Comment:
Conversely to my comment below, this seems like a good thing to turn into a `shared_ptr`, because there's nothing in the factory API which requires that the responder be created synchronously, right?